### PR TITLE
enhance: try compact small segments first if they may compose a full segment

### PR DESCRIPTION
**Pull request by tedxu**
See #37234

### CODE REVIEW

**Comment by tedxu** [comment]
rerun go-sdk

File: internal/datacoord/compaction_trigger.go [hunk 1]
```diff
@@ -548,107 +531,57 @@ func (t *compactionTrigger) generatePlans(segments []*SegmentInfo, signal *compa
...
+		if len(pack) == 0 {
+			break
 		}
-		return nonPlannedSegments[i].GetID() > nonPlannedSegments[j].GetID()
-	})
-
-	// greedy pick from large segment to small, the goal is to fill each segment to reach 512M
+		buckets = append(buckets, pack)
+	}
+	// 2. greedy pick from large segment to small, the goal is to fill each segment to reach 512M
```
Comment (XuanYang-cn) [inline internal/datacoord/compaction_trigger.go:554]: NIT: maxSize is dynamic， maybe donot mention specific size here

**Review by XuanYang-cn — COMMENTED** [review]
/lgtm

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**